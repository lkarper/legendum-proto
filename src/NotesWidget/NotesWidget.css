.NotesWidget__outer-container-widget-list.off {
    width: 0;
    height: 0;
    display: none;
}

.NotesWidget__outer-container-widget-list.on {
    width: calc(100vw - 10%);
    height: calc(100vh - 75px - 4%);
    word-wrap: break-word;
    position: fixed;
    top: 75px;
    margin: 5%;
    border: 1px solid black;
    background-color: white;
    z-index: 200;
    overflow: hidden;

}

.NotesWidget__toggle-button {
    position: fixed;
    right: 10px;
    top: 100px;
    z-index: 100;
}

.NotesWidget__outer-container-close-button {
    position: absolute;
    top: 3px;
    right: 3px;
}