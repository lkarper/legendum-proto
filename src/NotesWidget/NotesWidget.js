import React, { useState } from 'react';
import PropTypes from 'prop-types';
import TokenService from '../services/token-service';
import NotesList from '../NotesList/NotesList';
import './NotesWidget.css';

const NotesWidget = (props) => {
    const [showList, toggleShowList] = useState(false);

    const notesListContainerClassSuffix = showList ? 'widget-list on' : 'widget-list off';

    // Users must be logged in to save notes
    if (TokenService.hasAuthToken()) {
        return (
            <>
                <button
                    className='NotesWidget__toggle-button button'
                    onClick={() => toggleShowList(!showList)}
                >
                    My notes
                </button>
                <div className={`NotesWidget__outer-container-${notesListContainerClassSuffix}`}>
                    <button
                        className='NotesWidget__outer-container-close-button button'
                        onClick={() => toggleShowList(false)}
                    >
                        &#10006;
                    </button>
                    <NotesList 
                        chapt={props.match.params.chapt} 
                        suffix={'widget-list'} 
                    />
                </div>
            </>
        );
    }

    return <></>;
}

NotesWidget.defaultProps = {
    match: {
        params: {
            chapt: '',
        },
    },
};

NotesWidget.propTypes = {
    match: PropTypes.object,
};

export default NotesWidget;
