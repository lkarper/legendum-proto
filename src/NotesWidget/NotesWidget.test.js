import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import NotesWidget from './NotesWidget';

describe('NotesWidget component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<NotesWidget />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<NotesWidget />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
