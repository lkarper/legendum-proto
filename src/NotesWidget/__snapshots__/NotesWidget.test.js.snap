// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesWidget component renders the UI as expected 1`] = `
<NotesWidget
  match={
    Object {
      "params": Object {
        "chapt": "",
      },
    }
  }
/>
`;
